department-of-veterans-affairs/vets-api
# SLO Terms of Use decline

## Summary
This PR allows a new `agreements_declined=true` param to be passed on logout. When this is passed on Terms of Use decline and logout it will append  `appKey=https://dev-api.va.gov/agreements_declined`  param on the saml_ssoe logout_url -  `https://int.eauth.va.gov/slo/globallogout?appKey=https%253A%252F%252Fdev-api.va.gov%252Fagreements_declined`

Application Manifest PRs need to be merged first. 
- https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2383
- https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2384

`Settings.saml_ssoe.logout_url` with the `appKey` hardcoded is compatible with this version as the `appKey` will get overwritten. Once this is deployed the hardcoded `appKey` will be removed from the `logout_url`.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#68651
- https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2383
- https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2384


## How to test
- login with `oauth=false`
- GET [http://localhost:3000/v1/sessions/slo/new](http://localhost:3000/v1/sessions/slo/new) to logout
  - you should be redirected to `https://int.eauth.va.gov/slo/globallogout?appKey=https%253A%252F%252Fssoe-sp-dev.va.gov`

- login with `oauth=false`
- GET [http://localhost:3000/v1/sessions/slo/new?agreements_declined=true](http://localhost:3000/v1/sessions/slo/new?agreements_declined=true) to logout
   - You should be redirected to `https://int.eauth.va.gov/slo/globallogout?appKey=https%253A%252F%252Fdev-api.va.gov%252Fagreements_declined`

## What areas of the site does it impact?
SSOe logout/SLO
